opal/datatype/opal_datatype_optimize.c

```diff
@@ -310,5 +310,39 @@ int32_t opal_datatype_commit( opal_datatype_t * pData )
         pLast->first_elem_disp = first_elem_disp;
         pLast->size            = pData->size;
     }
+
+    /* generate iovec */
+    if( pData->iov == NULL )
+        opal_generate_iovec( pData );
+
     return OPAL_SUCCESS;
 }
+
+int32_t
+opal_generate_iovec( opal_datatype_t *pData )
+{
+    opal_convertor_t *local_convertor = opal_convertor_create( opal_local_arch, 0 );
+
+    size_t max = SIZE_MAX;
```

Comment:
You don't need to initialize max to SIZE_MAX, setting it to 0 should be good enough.